docs/_docs/persistence/change-data-capture.adoc

```diff
@@ -0,0 +1,128 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+= Change Data Capture
+
+WARNING: Experimental API.
+
+== Overview
+Change Data Capture (link:https://en.wikipedia.org/wiki/Change_data_capture[CDC]) is data processing pattern used to asynchronously receive entries that has changed on local node
+so that action can be taken using the changed entry.
+
+Some use-cases for CDC:
+
+ * Stream changes in Warehouse.
+ * Update search index.
+ * Calculate statistics (streaming queries).
+ * Audit logs.
+ * Async interaction with extenal system - moderation, business process invocation, etc.
+
+Ignite implements CDC with an application `ignite-cdc.sh` and link:https://github.com/apache/ignite/blob/master/modules/core/src/main/java/org/apache/ignite/cdc/CdcConsumer.java#L56[Java API].
+
+CDC application and Ignite node integrated via WAL archive segments:
+
+image:../../assets/images/integrations/CDC-design.svg[]
+
+When CDC enabled Ignite server node creates a hard link to each WAL archive segment in special `db/cdc/\{consistency_id\}` directory.
+`ignite-cdc.sh` run in different JVM and process newly archived link:native-persistence.adoc#_write-ahead_log[WAL segments].
+When segment fully processed by `ignite-cdc.sh` it removed. The actual disk space free when both links - archive and CDC removed.
+tate of consumption can be saved to disk to continue from it in the case of any failure.
```

Comment:
Personally, I would rework this sentence as it's not clear what the "State of consumption" concept is.

Reply:
> State of consumption is a pointer to the last processed event.
> Consumer can tell to `ignite-cdc.sh` to save the consumption state.
> On startup event processing will be continued from the last saved state.

Is it better?